usr.sbin/periodic/etc/daily/801.trim-zfs

```diff
@@ -0,0 +1,60 @@
+#!/bin/sh
+#
+#
+
+# If there is a global system configuration file, suck it in.
+#
+
+if [ -r /etc/defaults/periodic.conf ]
+then
+    . /etc/defaults/periodic.conf
+    source_periodic_confs
+fi
+
+case "$daily_trim_zfs_enable" in
+    [Yy][Ee][Ss])
+	echo
+	echo 'Trimming of zfs pools:'
+
+	if [ -z "${daily_trim_zfs_pools}" ]; then
+		daily_trim_zfs_pools="$(zpool list -H -o name)"
+	fi
+
+	rc=0
+	for pool in ${daily_trim_zfs_pools}; do
+		# sanity check
+		_status=$(zpool list -Hohealth "${pool}" 2> /dev/null)
+		if [ $? -ne 0 ]; then
+			rc=2
+			echo "   WARNING: pool '${pool}' specified in"
+			echo "            '/etc/periodic.conf:daily_trim_zfs_pools'"
+			echo "            does not exist"
+			continue
+		fi
+		case ${_status} in
+		FAULTED)
+			rc=3
+			echo "Skipping faulted pool: ${pool}"
+			continue ;;
+		UNAVAIL)
+			rc=4
+			echo "Skipping unavailable pool: ${pool}"
+			continue ;;
+		esac
+
+		if ! zpool status "${pool}" | grep -q '(trimming)'; then
+			echo "    starting trim of pool '${pool}'"
+			zpool trim "${pool}"
```

Comment:
Generally, we have knobs to control how the trim happens. For example `130.clean-msgs` has a parameter `daily_clean_msgs_days` which controls the number of days. Other things have flags that can be passed to this program or that.

There's only two options to zpool trim that might be relevant. One is `-d/--secure`  which would allow secure erase of the trimmed data. I'm not sure we even support that. The other is `-r/--rate` which specifies the maximum rate. That we support. Based on that, I'd add a `daily_trim_zfs_rate` variable and pass it -r to the zpool trim. You an see what `130.clean-msgs` does for an example. Had there been more options, I'd opt for the more generic `daily_trim_zfs_flags` to match the rc.d pattern, defaulting that to nothing and unconditionally passing that in.

Reply:
i'm inclined to do this with `daily_trim_zfs_flags` anyway, even if there's only a single useful option right now, since this means less work for users later if more flags are added - does that seem reasonable?